src/main/java/uk/gov/hmcts/darts/notification/service/impl/GovNotifyServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.darts.notification.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.notification.dto.GovNotifyRequest;
+import uk.gov.hmcts.darts.notification.service.GovNotifyService;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+import uk.gov.service.notify.SendEmailResponse;
+
+@RequiredArgsConstructor
+@Service
+@Slf4j
+public class GovNotifyServiceImpl implements GovNotifyService {
+
+    @Value("${darts.notification.gov-notify.api-key}")
+    private String apiKey;
+
+    @Override
+    public SendEmailResponse sendNotification(GovNotifyRequest request) throws NotificationClientException {
+
+        log.trace("Sending email with following settings = {}", request);
+        NotificationClient client = new NotificationClient(apiKey);
```

Comment:
So you can expose any object as a bean, you need to create a bean config file something like:

```
@Configuration
public class BeanConfiguration {

    @Value("${darts.notification.gov-notify.api-key}")
    private String apiKey;

    @Bean
    public NotificationClient notificationClient() {
        return new NotificationClient(apiKey);
    }
}
```

Then in the GovNotifyServiceImpl just autowire the NotificationClient.  The only reason not to do this would be if the NotificationClient isnt thread safe.

Reply:
Thanks, done